src/FakeItEasy/Creation/DefaultFakeAndDummyManager.cs

```diff
@@ -48,19 +52,18 @@ public bool TryCreateDummy(Type typeOfDummy, out object result)
             return false;
         }
 
-        private static IFakeOptions CreateFakeOptions(Type typeOfFake, ProxyOptions proxyOptions)
+        private static ConstructorInfo GetFakeOptionsConstructor(Type typeOfFake)
         {
-            var optionsConstructor = typeof(FakeOptions<>)
+            return typeof(FakeOptions<>)
                 .MakeGenericType(typeOfFake)
                 .GetConstructor(new[] { typeof(ProxyOptions) });
-
-            return (IFakeOptions)optionsConstructor.Invoke(new object[] { proxyOptions });
         }
 
         private IProxyOptions BuildProxyOptions(Type typeOfFake, Action<IFakeOptions> optionsBuilder)
         {
             var proxyOptions = new ProxyOptions();
-            var options = CreateFakeOptions(typeOfFake, proxyOptions);
+            var constructor = FakeOptionsConstructorCache.GetOrAdd(typeOfFake, GetFakeOptionsConstructor);
+            var options = (IFakeOptions) constructor.Invoke(new object[] { proxyOptions });

```

Comment:
I think we can do better than that. Instead of caching and invoking the constructor, create a delegate, cache it and invoke it. It would probably be faster.

Reply:
Just did a quick test, the delegate is about 8 times faster. And it doesn't require an array allocation.